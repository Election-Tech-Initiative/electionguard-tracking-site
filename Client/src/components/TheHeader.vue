<template>
    <header class="header">
        <div class="container">
            <div class="is-flex">
                <img class="election-logo" src="/logo.png" :alt="$t('logoAltText', { location: appSettings.locationDescription })">
                <span class="title">{{pageTitle}}</span>
            </div>
        </div>
    </header>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class TheHeader extends Vue {
    @Prop() public readonly pageTitle!: string;
}
</script>

<style lang="scss" scoped>

.is-flex {
    align-items: center;
    @include touch {
        margin: 0 1rem;
    }
}

.election-logo {
  width: 10rem;
  height: 10rem;

  @include mobile {
      width: 6rem;
      height: 6rem;
  }
}

.title {
    font-size: 2.2rem;
    margin-left: 2.6rem;
    @include mobile {
        font-size: 1.7rem;
        margin-left: 0.5rem;
    }
}

</style>
