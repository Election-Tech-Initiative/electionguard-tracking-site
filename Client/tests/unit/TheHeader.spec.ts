import { mount } from '@vue/test-utils';

import TheHeader from '@/components/TheHeader.vue';

describe('TheHeader.vue', () => {
    it('renders', () => {
        const wrapper = mount(TheHeader, {
            mocks: {
                appSettings: {
                    locationDescription: 'Mocked Location',
                },
                $t: (msg: string): string => msg
            },
            propsData: {
                pageTitle: 'Page Title'
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
