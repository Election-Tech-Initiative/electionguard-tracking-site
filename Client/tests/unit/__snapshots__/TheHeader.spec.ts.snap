// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader.vue renders 1`] = `
<header
  class="header"
>
  <div
    class="container"
  >
    <div
      class="is-flex"
    >
      <img
        alt="logoAltText"
        class="election-logo"
        src="/logo.png"
      />
       
      <span
        class="title"
      >
        Page Title
      </span>
    </div>
  </div>
</header>
`;
